cmd/algokey/multisig.go

```diff
@@ -93,3 +105,78 @@ var multisigCmd = &cobra.Command{
 		}
 	},
 }
+
+var convertCmd = &cobra.Command{
+	Use:   "convert -t [transaction file] -p \"[threshold] [PK1] [PK2] ...\"",
+	Short: "Adds the necessary fields to a transaction that is sent from an account to was rekeied to a multisig account.",
+	Args:  cobra.NoArgs,
+	Run: func(cmd *cobra.Command, _ []string) {
+
+		// Read Transaction
+		txdata, err := ioutil.ReadFile(multisigTxfile)
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "Cannot read transactions from %s: %v\n", multisigTxfile, err)
+			os.Exit(1)
+		}
+
+		var outBytes []byte
+		dec := protocol.NewDecoderBytes(txdata)
+
+		var stxn transactions.SignedTxn
+		err = dec.Decode(&stxn)
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "Cannot decode transaction: %v\n", err)
+			os.Exit(1)
+		}
+
+		// Decode params
+		params := strings.Split(msigParams, " ")
+		if len(params) < 2 {
+			_, _ = fmt.Fprint(os.Stderr, "Not enough arguments to create the multisig address.\nPlease make sure to specify the threshold and addresses")
+			os.Exit(1)
+		}
+
+		threshold, err := strconv.ParseUint(params[0], 10, 8)
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "Failed to parse the threshold. Make sure it's a number between 1 and 255: %v\n", err)
+			os.Exit(1)
+		}
+
+		// convert addresses to pks
+		// convert the addresses into public keys
+		pks := make([]crypto.PublicKey, len(params[1:]))
+		for i, addrStr := range params[1:] {
+			addr, err := basics.UnmarshalChecksumAddress(addrStr)
+			if err != nil {
+				fmt.Fprintf(os.Stderr, "Cannot decode address: %v\n", err)
+				os.Exit(1)
+			}
+			pks[i] = crypto.PublicKey(addr)
+		}
+
+		addr, err := crypto.MultisigAddrGen(1, uint8(threshold), pks)
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "Cannot generate multisig addr: %v\n", err)
+			os.Exit(1)
+		}
+
+		//gen the multisig and assign to the txn
+		stxn.Msig = crypto.MultisigPreimageFromPKs(1, uint8(threshold), pks)
+
+		//append the signer since it's a rekey txn
+		if basics.Address(addr) == stxn.Txn.Sender {
+			fmt.Fprintf(os.Stderr, "The sender at the msig address should not be the same: %v\n", err)
+			os.Exit(1)
+		}
+		stxn.AuthAddr = basics.Address(addr)
+
+		// Write the txn
+		outBytes = append(outBytes, protocol.Encode(&stxn)...)
+
+		err = ioutil.WriteFile(multisigTxfile, outBytes, 0600)
```

Comment:
maybe use `writeFile` so that we can support `stdoutFilenameValue` ?

Reply:
I'm not sure I follow. It does use `WriteFile`. Do you mean to let the user choose the name?